pkg/proxy/healthcheck/healthcheck_test.go

```diff
@@ -369,26 +369,27 @@ func TestHealthzServer(t *testing.T) {
 	fakeClock := clock.NewFakeClock(time.Now())
 
 	hs := newProxierHealthServer(listener, httpFactory, fakeClock, "127.0.0.1:10256", 10*time.Second, nil, nil)
-	server := hs.httpFactory.New(hs.addr, healthzHandler{hs: hs})
+	healthSvr := hs.(*proxierHealthServer)
+	server := healthSvr.httpFactory.New(healthSvr.addr, healthzHandler{hs: healthSvr})
```

Comment:
rather than doing this, just have the internal method (`newProxierHealthServer`) still return a `*ProxierHealthServer`, and only change the public method (`NewProxierHealthServer`) to return `ProxierHealthUpdater`